spec-0007/transition_to_rng.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import functools
+import warnings
+
+
+def _transition_to_rng(old_name, position_num=None, dep_version=None):
+    """Example decorator to transition from old PRNG usage to new `rng` behavior
+
+    Suppose the decorator is applied to a function that used to accept parameter
+    `old_name='random_state'` either by keyword or as a positional argument at
+    `position_name=1`. At the time of application, the name of the argument in the
+    function signature is manually changed to the new name, `rng`. If positional
+    use was allowed before, this is not changed.*
+
+    - If the function is called with both `random_state` and `rng`, the decorator
+      raises an error.
+    - If `random_state` is provided as a keyword argument, the decorator passes
+      `random_state` to the function's `rng` argument as a keyword, but the decorator
+      will emit a `DeprecationWarning` about the deprecation of keyword `random_state`.
+    - If `random_state` is provided as a positional argument, the decorator passes
+      `random_state` to the function's `rng` argument by position, but the decorator
+      will emit a `FutureWarning` about the changing interpretation of the argument.
```

Comment:
Technically if a `Generator` is passed by position or if `None` is passed by position and `np.random.seed` has not been called, we don't need to warn. Interpretation of those things won't change. LMK if the additional logic for sparing the warning in those cases should be added.

Reply:
Addressed by https://github.com/stefanv/specs/commit/636bc5d1c128b94dfefe9014882adade63b570d3